nixos/modules/services/networking/sabnzbd/default.nix

```diff
@@ -0,0 +1,605 @@
+        if true == v then
+          "1"
+        else if false == v then
+          "0"
+        else
+          mkValueStringDefault { } v
+      );
+      mkKeyValue = mkKeyValueDefault { inherit mkValueString; } "=";
+      mkSection = (
+        depth: attrs:
+        let
+          atoms = extractAtoms attrs;
+          sections = extractSections attrs;
+          sectionHeadingLeft = lib.concatStrings (lib.replicate (depth + 1) "[");
+          sectionHeadingRight = lib.concatStrings (lib.replicate (depth + 1) "]");
+          mkSectionHeading =
+            name: "${sectionHeadingLeft}${lib.escape [ "[" "]" ] name}${sectionHeadingRight}";
+          mkSubsection = name: attrs: (mkSectionHeading name) + "\n" + (mkSection (depth + 1) attrs) + "\n";
+        in
+        toKeyValue { inherit mkKeyValue; } (extractAtoms attrs)
+        + "\n"
+        + lib.concatStrings (lib.mapAttrsToList mkSubsection sections)
+      );
+    in
+    {
+      type = types.attrsOf configObjValue;
+      generate = name: attrs: pkgs.writeText name (mkSection 0 attrs);
+    };
+
+  publicSettingsIni =
+    if cfg.configFile != null then
+      cfg.configFile
+    else
+      (configObjIni { }).generate "public-settings.ini" allSettings;
+
+  sabnzbdIniPath = "/var/lib/${cfg.stateDir}/sabnzbd.ini";
+in
+
+{
+  options = {
+    services.sabnzbd = {
+      enable = mkEnableOption "the sabnzbd server";
+
+      package = mkPackageOption pkgs "sabnzbd" { };
+
+      configFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = "Path to config file (deprecated, use `settings` instead)";
+      };
+
+      stateDir = mkOption {
+        type = types.str;
+        default = "sabnzbd";
+        description = "State directory of the service under /var/lib/";
+      };
+
+      user = mkOption {
+        default = "sabnzbd";
+        type = types.str;
+        description = "User to run the service as";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "sabnzbd";
+        description = "Group to run the service as";
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Open ports in the firewall for the sabnzbd web interface
+        '';
+      };
+
+      secretsFile = mkOption {
+        type = types.nullOr types.path;
+        description = ''
+          Path to an ini file containing confidential settings such as credentials.
+          Settings here will be merged with the rest of the configuration (with
+          the secret settings taking precedence in case of conflicts).
+          Recommended settings:
+          - misc.api_key, misc.nzb_key, misc.username, misc.password
+          - misc.email_account, misc.email_pwd if email alerts are enabled
+          - servers.<name>.username, servers.<name>.password
+        '';
+        default = null;
+      };
+
+      allowConfigWrite = mkOption {
```

Comment:
Is this not a breaking change for people who have already made changes through the web interface?

Reply:
Good catch, thanks! Should this be conditional on `system.stateVersion` then?